app/controllers/application_controller.rb

```diff
@@ -247,6 +249,117 @@ def initiate_wait_for_task(options = {})
   end
   private :initiate_wait_for_task
 
+  # Method for creating object with data for report.
+  # Report is either grid/table or list.
+  # @param controller_name name of JS controller. Typically `reportDataController`.
+  def init_report_data(controller_name)
+    view_url = view_to_url(@view) unless @view.nil?
+    {
+      :controller_name => controller_name,
+      :data            => {
+        :modelName  => @display.nil? && !self.class.model.nil? ? self.class.model.to_s.tableize : @display,
+        :activeTree => x_active_tree.to_s,
+        :gtlType    => @gtl_type,
+        :currId     => params[:id],
+        :sortColIdx => @sortcol,
+        :sortDir    => @sortdir,
+        :isExplorer => @explorer,
+        :showUrl    => view_url
+      }
+    }
+  end
+
+  # Private method for processing params.
+  # params can contain these options:
+  # @param params parameters object.
+  # @option params :explorer [String]
+  #     String value of boolean if we are fetching data for explorer or not. "true" | "false"
+  # @option params :active_tree [String]
+  #     String value of active tree node.
+  # @option params :model_id [String]
+  #     String value of model's ID to be filtered with.
+  def process_params_options(params)
+    options = {}
+    @explorer = params[:explorer] == "true" if params[:explorer]
+
+    if params[:active_tree]
+      node_info = get_node_info(x_node, false) if defined? get_node_info
```

Comment:
get_node_info might not be defined for current controller, hence the `defined? get_node_info`, however I think it might be better to move it to upper `if params[:active_tree]`